### PR TITLE
Add test for formatting issue

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

Adds test for the fix:
https://github.com/cowprotocol/cowswap/pull/2501

This PR is done separately from the FIX because I wanted to do the changes on top of the refactor

### CODE REVIEW

File: src/utils/amountFormat/index.test.ts [hunk 1]
```diff
@@ -27,6 +27,30 @@ describe('Amounts formatting', () => {
...
+      const result = formatTokenAmount(getAmount('10000100', -5)) // 100.00100
+
+      expect(result).toBe('100.001')
+    })
+
+    it('Trim all trailing zero', () => {
+      const result = formatTokenAmount(getAmount('10000000', -5)) // 100.00000
+
+      expect(result).toBe('100')
+    })
```
Comment (anxolin) [inline src/utils/amountFormat/index.test.ts:46]: The cases above where failing before the fix

**Comment by elena-zh** [comment]
Hey @anxolin , looks like the issue with ` .0000` is still here
image

**Review by alfetopito — APPROVED** [review]

File: src/common/pure/TokenAmount/index.tsx [hunk 2]
```diff
@@ -53,9 +53,10 @@ export function TokenAmount({
       </>
     )
 
+  const roundedAmount = round ? FractionUtils.round(amount) : amount
   return (
     <Wrapper title={title} className={className} highlight={highlight}>
```
[the remark this answers is not in the document]
Comment (anxolin) [reply src/common/pure/TokenAmount/index.tsx]: I think it's fine the title show the exact full amount, and the amount just rounds up.

**Comment by anxolin** [comment]
@elena-zh i tried to reproduce what you received in the screenshot by creating these tests:

image


As you can see, here is trimming. I used USDC which is also 6 decimals token. 

Maybe this was fixed in the hotfix? 
I will merge this PR since it only includes new tests and documentation.

If this issue is still there, let me know. We can try to recreate the scenario

**Pull request merged**